packages/ui/src/common/OTPField.tsx

```diff
@@ -0,0 +1,68 @@
+import * as Common from '@clerk/elements/common';
+import React from 'react';
+
+import * as Field from '../primitives/field';
+
+export function OTPField({
+  label = 'One-time password',
+  ...props
+}: React.ComponentProps<typeof Common.Input> & {
+  label?: React.ReactNode;
+}) {
+  return (
+    <Common.Field
+      asChild
+      name='code'
+    >
+      <Field.Root>
+        <Common.Label asChild>
+          <Field.Label visuallyHidden>{label}</Field.Label>
+        </Common.Label>
+
+        <Common.FieldState>
+          {({ state }) => (
+            <Common.Input
+              type='otp'
+              autoSubmit
+              className='flex gap-2'
+              render={({ value, status }) => (
+                <Field.Input
+                  type='otp'
+                  data-status={status}
+                  intent={state}
+                  state={
+                    (
+                      {
+                        cursor: 'focus-visible',
+                        selected: 'focus-visible',
+                        hovered: 'hover',
+                        none: 'native',
+                      } as const
+                    )[status]
+                  }
+                  {...props}
+                  asChild
+                >
+                  <span>{value}</span>
```

Comment:
Can we implement the cursor when the segment is in the cursor status?

Reply:
Could be misunderstanding but I thought I achieved this here?
https://github.com/clerk/javascript/pull/3535/files#r1632939660

Reply:
oh right or are you saying we should have a blinking cursor to mimic other inputs?

Reply:
Done!